### ISSUE TITLE
Missing steps to get tch-rs to work on windows?

### ISSUE THREAD

**Issue by junaid-nom** [LaurentMazare/tch-rs#534]
Sorry if this is a dumb sequence of questions but I've spent hours trying to get this to work on windows.

I added the environment variables as per the README. The libtorch\lib directory to PATH and LIBTORCH to libtorch\

I then had build issues with cc-rust dependency and some others saying there was no g++. I resolved that by installed mysys2 and mingw.

But then libtorch fails to compile with:
```
error occurred: Command "g++.exe" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "C:\\Users\\xjuna\\Documents\\pytorch\\libtorch\\include" "-I" "C:\\Users\\xjuna\\Documents\\pytorch\\libtorch\\include/torch/csrc/api/include" "-o" "C:\\Users\\xjuna\\Documents\\live_to_want\\deep_ai\\target\\debug\\build\\torch-sys-b2c8cfa865438e10\\out\\libtch/torch_api.o" "-c" "libtch/torch_api.cpp" with args "g++.exe" did not execute successfully (status code exit code: 1).
```

I believe its because pytorch does not support the mingw compiler?

So I remove the mingw env variable. Then I try to use MSVC. I do this by booting up the developer powershell for VS that has the C++ package installed. I then set the env variable CXX="CL" to try and get it to use that for the c++ compiler, grasping at straws really, not familiar with C++ compiling especially on windows.

But then I get the error:
`pytorch\libtorch\include\c10/util/safe_numerics.h(25): error C3861: '_addcarry_u64': identifier not found`

Am I missing something here? Is the install supposed to be this difficult for windows?

Am I supposed to compile the pytorch library itself first somehow?

Also I downloaded pytorch C++ 1.12.1 not 1.12.0 but since its a minor version I didn't think it would make a difference?

**Comment by joverwey** [comment]
Which rust toolchain are you using? I'm using `stable-x86_64-pc-windows-msvc` on Windows 11 and haven't had any problems. I didn't have to set any special env variables except the ones explained in the setup README.md. And that is only if you need CUDA. Try without CUDA and local install of libtorch first just to get it up and running.

**Comment by junaid-nom** [comment]
Omg changing the toolchain worked!
All I had to do was run:
`rustup default stable-x86_64-pc-windows-msvc`
AHHH I wasted so many hours doing all sorts of weird stuff trying to get this to work!

I would really appreciate it and want to save future noobs like myself a lot of hours if they added that to the readme somewhere. Just a single "Make sure you are using the msvc rust toolchain which can be installed via rustup". Something like that.
